lib/buffer.js

```diff
@@ -270,6 +270,11 @@ function fromObject(obj) {
       }
       return fromArrayLike(obj);
     }
+    if (typeof SharedArrayBuffer !== 'undefined') {
```

Comment:
Since the `fromObject` method was being called in `Buffer.from` I thought it would be okay to place it there instead. 
But yes I realized that mistake quite late, I mentioned something similar in #8440.
So I guess I should move it to `Buffer.from` and have `fromArrayBuffer` handle it?